cli/admin/call.go

```diff
@@ -0,0 +1,187 @@
+package admin
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/tarantool/cartridge-cli/cli/project"
+
+	"github.com/spf13/pflag"
+
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"github.com/tarantool/cartridge-cli/cli/templates"
+)
+
+const (
+	argTypeString = "string"
+	argTypeNumber = "number"
+	argTypeBool   = "boolean"
+)
+
+type FuncCallArg struct {
+	Name string
+	Type string
+
+	StringValue string
+	NumberValue int
+	BoolValue   bool
+
+	Changed bool
+}
+
+func adminFuncCall(conn net.Conn, funcName string, flagSet *pflag.FlagSet, args []string) error {
+	funcCallArgs, err := getFuncCallArgs(conn, funcName, flagSet, args)
+	if err != nil {
+		return fmt.Errorf("Failed to parse function call args: %s", err)
+	}
+
+	argsSerialized, err := serializeArgs(funcCallArgs)
+	if err != nil {
+		return fmt.Errorf("Failed to serialize function args: %s", err)
+	}
+
+	callFuncBody, err := templates.GetTemplatedStr(&adminCallFuncBodyTmpl, map[string]string{
+		"AdminCallFuncName": adminCallFuncName,
+		"FuncName":          funcName,
+		"Args":              argsSerialized,
+	})
+
+	callResRaw, err := common.EvalTarantoolConn(conn, callFuncBody)
+	if err != nil {
+		return fmt.Errorf("Failed to call %q: %s", funcName, err)
+	}
+
+	needReturnValueWarn := false
+
+	switch callRes := callResRaw.(type) {
+	case string:
+		log.Info(callRes)
```

Comment:
Is it debug?

Reply:
No, it's how the function result is logged.

Reply:
It's not clear from code. Please add some comment or move to separate function.

Reply:
Moved to function.